packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -457,4 +474,268 @@ linter:
       printStatus('No analyzer warnings! (ran in ${elapsed}s)');
     return 0;
   }
+
+  Future<int> _analyzeWatch() async {
+    List<String> directories;
+
+    if (isFlutterRepo) {
+      printStatus('Analyzing Flutter repository:');
+      directories = <String>[];
+      directories.addAll(_gatherProjectPaths(path.absolute('examples')));
+      directories.addAll(_gatherProjectPaths(path.absolute('packages')));
+      for (String projectPath in directories)
+        printStatus('  ${path.relative(projectPath)}');
+      printStatus('');
+    } else {
+      directories = <String>[Directory.current.path];
+    }
+
+    AnalysisServer server = new AnalysisServer(dartSdkPath, directories);
+    server.onAnalyzing.listen(_handleAnalysisStatus);
+    server.onErrors.listen(_handleAnalysisErrors);
+
+    await server.start();
+
+    int exitCode = await server.onExit;
+    printTrace('Analysis server exiting with code $exitCode.');
+    return 0;
+  }
+
+  bool firstAnalysis = true;
+  Set<String> analyzedPaths = new Set<String>();
+  Map<String, List<AnalysisError>> analysisErrors = <String, List<AnalysisError>>{};
+  Stopwatch analysisTimer;
+  int lastErrorCount = 0;
+
+  void _handleAnalysisStatus(bool isAnalyzing) {
+    if (isAnalyzing) {
+      if (firstAnalysis) {
+        printStatus('Analyzing ${path.basename(Directory.current.path)}...');
+      } else {
+        printStatus('');
+      }
+
+      analyzedPaths.clear();
+      analysisTimer = new Stopwatch()..start();
+    } else {
+      analysisTimer.stop();
+
+      // Sort and print errors.
+      List<AnalysisError> errors = <AnalysisError>[];
+      for (List<AnalysisError> fileErrors in analysisErrors.values)
+        errors.addAll(fileErrors);
+
+      errors.sort();
+
+      for (AnalysisError error in errors)
+        printStatus(error.toString());
```

Comment:
Shouldn't this go through the same post-analyzer processing as one-shot analysis?